samples/net/sockets/echo_server/src/echo-server.c

```diff
@@ -84,6 +84,9 @@ static void stop_udp_and_tcp(void)
 static void event_handler(struct net_mgmt_event_callback *cb,
 			  uint32_t mgmt_event, struct net_if *iface)
 {
+	ARG_UNUSED(iface);
```

Comment:
The changes in `echo_server` sample seem a bit unrelated to other work in this commit, perhaps they merit a separate commit?

Reply:
I suppose these changes could be moved to a commit between this one and the commit before it -- But I included them in this commit since this is technically the commit that causes these changes to be required. Although in reality, the changes would work and are justifiable as early as hyr/pull/56549/commits/b8cf864359676657057323c35299cb533dbfdbc8

I'm pretty neutral, so final decision is up to you

Reply:
Ah, after having a second look I now understand why are the changes related. I think that it could either be extracted into a separate commit after b8cf864, or, at very least, you could elaborate in the commit message on why are the changes needed.

Reply:
I'll do both, extract as a separate commit, and I'll try to update all the commit messages to include a bit more why, as per @jukkar 's very good point here: https://github.com/zephyrproject-rtos/zephyr/pull/57057#discussion_r1172540424